### PR TITLE
support adding and pushing git notes

### PR DESCRIPTION
**Pull request by ahume**
We needed this to support some use cases with how our pipelines integrate with other systems. Thought I'd push back upstream to see if it was of interest. If not, a quick sanity check of the code would be fantastic. Test image here...

andyhume/git-resource-with-notes:v0.0.1

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@ahume Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/git-resource&pullRequestId=103)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=concourse/git-resource&pullRequestId=103) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@ahume Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/git-resource&pullRequestId=103)!

File: assets/out [hunk 1]
```diff
@@ -92,6 +98,13 @@ push_tags() {
   git push --tags push-target $forceflag
 }
 
+add_and_push_notes() {
+  if [ -n "$notes" ]; then
+    git notes add -m "${notes}"
```
Comment (vito) [inline assets/out]: This would be cleaner as `git nodes add -F $notes_file` rather than `cat`ing it into a variable

Comment (ahume) [inline assets/out]: Hm, I think I was thrown by the fact I'd need to reference the file outside of the repo. i.e.

`git notes add -F "../${notes_file}"`

What do you think?

Comment (ahume) [inline assets/out]: Done as you suggested @vito - agree it feels better.

How is this PR looking now?

**Comment by ahume** [comment]
Bounce.

@vito What do you want to do with this? Happy to close it if notes support isn't needed/wanted.

**Pull request merged**